### ISSUE TITLE
bug(populate): missing childKey causes no population

### ISSUE THREAD

**Issue by raghavendracs** [prescottprue/react-redux-firebase#277]
**Do you want to request a *feature* or report a *bug*?**  

*bug*

**What is the current behavior?**

Trying with 2.0.0 latest version.

When trying to pass a set of child keys to populate if some of the keys does not exist in the firebase route none of the keys are populated.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

I have a /users/uid route in firebase which has only "experiences" key. If non existent childkeys are included in the populates list existing keys should still be populated. 

Populate works fine when I try to initiate with 

```js 
const populates = [
  { child: 'experiences', root: '/allExperiencesList' },
]

const fbWrappedComponent = firebaseConnect(
  ({AuthProfile}) => {
    if(AuthProfile){
      return ([{path: `/users/${AuthProfile ? AuthProfile.uid : ''}`, populates}])
    }
  }
)(TopBarContainer)
```
but fails if I add a key which does not exist and try as follows,

```js
const populates = [
  { child: 'experiences', root: '/allExperiencesList' },
  { child: 'education', root: '/allEducationList' },
]

const fbWrappedComponent = firebaseConnect(
  ({AuthProfile}) => {
    if(AuthProfile){
      return ([{path: `/users/${AuthProfile ? AuthProfile.uid : ''}`, populates}])
    }
  }
)(TopBarContainer)
```
Tried enabling logging to "true" and clearly see that populate is not even triggered in the second case.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**
As far as I know it did work in previous versions.

**Comment by prescottprue** [comment]
Going to attempt to replicate, but a few things that stick out initially:

1. If you do not use connect, you will never be getting this data out of redux (maybe it got left out of the example or I missed something?)
1. If`AuthProfile`doesn't exist, then nothing is returned to `firebaseConnect`, which means that no queries will be created
1. Your queries might work more as expected if you [load/check Auth in a higher component and pass it down](https://github.com/prescottprue/react-redux-firebase/blob/master/examples/snippets/stateBasedQuery/App.js#L14-L25) as done in the [state based query example](https://github.com/prescottprue/react-redux-firebase/blob/master/examples/snippets/stateBasedQuery/App.js)
1. It may not change things at all, but `root` values shouldn't need `/` (could cause issue, not sure)

**Side Note** May be totally wrong but, I believe that capitalizing prop names is seen as frowned upon

**Comment by raghavendracs** [comment]
Thanks Scott. 

1. Yes there indeed a connect as well just did not include in the snippet.

2,3. This component indeed is launched only for logged in users

4. Sure realised that and had tried without the / but situation does not change.

Also will look at de-capitalising the prop names if that is standard approach.

**Comment by prescottprue** [comment]
If the component only shows for authenticated users, is the if statement within the `firebaseConnect` necessary (assuming `AuthProfile` prop) carries the auth. That could keep queries from being created.

**Comment by raghavendracs** [comment]
No that is surely not needed. Was trying various things and it must have been left behind. I see same behaviour even without that check.

________________________________

**Comment by prescottprue** [comment]
@raghavendracs Is it possible for you to include the full example (including those `connect` snippets) so that the issue can be replicated?

Are you attempting to populate your own user profile? If so, it would take using `profileParamsToPopulate` along with `populate`.

**Comment by prescottprue** [comment]
@raghavendracs Would love to reproduce the bug so it can get fixed. Any way you can post your full example?

**Comment by raghavendracs** [comment]
Sure Scott. Will pull out the relevant code from my repo and post.

Best Regards,
Raghu

+44 7470 123 259
________________________________

**Comment by pzmudzinski** [comment]
I am able to reproduce it on v2. 
This is my config:
```
export const profilePopulateSettings = [
  { child: 'neighborhood', root: 'neighborhoods' },
  { child: 'directMessages', root: 'directMessages' },
];
```

Each user has his neighborhood, but not all of them have direct messages. 
Something like:  
```
users: {
 user1: {
  neighborhood: <key>,
  directMessages: <optional object>
 }
}
```

I pass that into a `reactReduxFirebase` as a config: 
```
const config = {
  userProfile: 'users', // firebase root where user profiles are stored
  enableLogging: false, // enable/disable Firebase's database logging
  profileParamsToPopulate: profilePopulateSettings,
};
```

Then in my container I have something like that: 
```  
connect(
    state => ({
      auth: state.firebase.auth,
      profile: populate(state.firebase, 'profile', profilePopulateSettings) // third argument does not matter,
    }),
  ),
```

If user has both `neighborhood` and `directMessages` then profile is populated. If `directMessages` are missing there's *nothing* populated - even non-empty `neighborhood`.
Profile gets populated only if both properties exist. 
Expected - `profile.neighborhood` is expanded, `directMessages` stay empty.

**Comment by prescottprue** [comment]
@pzmudzinski thanks for pointing out a easy to reproduce case. Going to look into this later this week, but always open to PRs.

**Comment by prescottprue** [comment]
Fix was included as part of [`v2.0.0-beta.11`](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.0.0-beta.11). 

Shout out to @jeremyplease for the PR that fixed it
